internal/services/mssql/mssql_database_resource.go

```diff
@@ -213,6 +213,8 @@ func resourceMsSqlDatabaseCreate(d *pluginsdk.ResourceData, meta interface{}) er
 	location := server.Model.Location
 	ledgerEnabled := d.Get("ledger_enabled").(bool)
 
+	// This does not make sense here, it will always be 404 since the database does not exist?
```

Comment:
I'm not sure what this comment is referring to. Should we remove it or expand on it?

Reply:
It's because this call always returns a 404 because we look for a partner database before we even create it. Since this was a direct port from the existing code I thought I would call it out. It appears that this code is not needed, else there is some scenario I am unaware of where it would not return a 404.